tools/build_defs/cc_toolchain_util.bzl

```diff
@@ -58,20 +56,12 @@ def _to_depset(element):
     return depset(element)
 
 def _configure_features(ctx, cc_toolchain):
-    if (len(BAZEL_VERSION) == 0 or
-        versions.is_at_least("0.25.2", BAZEL_VERSION)):
-        return cc_common.configure_features(
-            ctx = ctx,
-            cc_toolchain = cc_toolchain,
-            requested_features = ctx.features,
-            unsupported_features = ctx.disabled_features,
-        )
-    else:
-        return cc_common.configure_features(
-            cc_toolchain = cc_toolchain,
-            requested_features = ctx.features,
-            unsupported_features = ctx.disabled_features,
-        )
+    return cc_common.configure_features(
```

Comment:
Since we no longer officially support versions below `3.4.0` I figured this was safe to remove. This was the primary justification for me wanting to remove `BAZEL_VERSION`